dev-client/src/components/home/SiteMap.tsx

```diff
@@ -30,30 +35,92 @@ const siteFeatureCollection = (
   })),
 });
 
-const SiteCallout = ({site}: {site: Site}): JSX.Element => {
+type SiteCalloutProps = {
+  site: Site;
+  closeCallout: () => void;
+};
+const SiteCallout = ({site, closeCallout}: SiteCalloutProps) => {
+  const {t} = useTranslation();
+  const project = useSelector(state =>
+    site.projectId === undefined
+      ? undefined
+      : state.project.sites[site.projectId],
+  );
+
   return (
     <Mapbox.MarkerView
       coordinate={[site.longitude, site.latitude]}
       anchor={{x: 0.5, y: 0}}
       allowOverlap={true}>
-      <Box backgroundColor="background.default">
-        <Heading size="md">{site.name}</Heading>
-        <Text>latitude: {site.latitude}</Text>
-        <Text>longitude: {site.longitude}</Text>
+      <Box bg="grey.200" padding="4">
+        <Flex direction="row" align="top" justify="space-between">
+          <Heading size="lg">{site.name}</Heading>
+          <MaterialIconButton
+            name="close"
+            onPress={closeCallout}
+            iconProps={{size: 'md'}}
+          />
+        </Flex>
+        {project && <Heading size="md">{project.name}</Heading>}
+        <Box height="4" />
+        <Box>
+          <Text>
+            {t('site.last_updated', {
+              date: 'dd-mm-yyyy',
+            })}
+          </Text>
+          <Text>{t('site.progress', {progress: '??'})}</Text>
+          <Flex direction="row">
+            <Badge>{t('site.members', {members: 'x'})}</Badge>
+          </Flex>
+        </Box>
       </Box>
     </Mapbox.MarkerView>
   );
 };
 
-const TemporarySiteCallout = ({site}: {site: Site}): JSX.Element => {
+const CalloutDetail = ({label, value}: {label: string; value: string}) => {
+  return (
+    <Box>
+      <Text bold>{label}:</Text>
+      <Text>{value}</Text>
+    </Box>
+  );
+};
+
+const TemporarySiteCallout = ({site, closeCallout}: SiteCalloutProps) => {
+  const {t} = useTranslation();
+  const {navigate} = useNavigation<TopLevelNavigationProp>();
+
+  const onCreate = useCallback(
+    () => navigate(ScreenRoutes.CREATE_SITE, {mapCoords: site}),
+    [site, navigate],
+  );
+
   return (
     <Mapbox.MarkerView
       coordinate={[site.longitude, site.latitude]}
       anchor={{x: 0.5, y: 0}}
       allowOverlap={true}>
-      <Box backgroundColor="background.default">
-        <Text>latitude: {site.latitude} </Text>
-        <Text>longitude: {site.longitude}</Text>
+      <Box backgroundColor="background.default" padding="4">
+        <Flex direction="row" align="top" justify="space-between">
+          <CalloutDetail label={t('site.soil_id_prediction')} value="???" />
+          <MaterialIconButton name="close" onPress={closeCallout} />
+        </Flex>
+        <Divider />
+        <CalloutDetail
+          label={t('site.ecological_site_prediction')}
+          value="???"
```

Comment:
question: what is the value supposed to be?

Reply:
some output of the soil id algorithm which doesn't exist yet